ably/realtime_conn.go

```diff
@@ -306,7 +307,7 @@ func (c *Connection) connectWithRetryLoop(arg connArgs) (result, error) {
 
 	// The initial DISCONNECTED event has been fired. If we reach stateTTL without
 	// any state changes, we transition to SUSPENDED state
-	stateTTL := c.opts.connectionStateTTL()
+	stateTTL := c.connectionStateTTL()
```

Comment:
Sorry, I don't see your point. My point is that the field `connStateTTL` that this PR introduces is always the same as `connDetails.ConnectionStateTTL`, so it isn't useful, since we can always use the latter. In fact, `connectionStateTTL` method's whole purpose seem to be to fall back to the latter when the former isn't set, which would never happen, so it's not useful.

Reply:
No, if you look at the function implementation, it's reading connectionTTL from different sources, they can vary based on whether `connected` message sets it or not